cmd/calyptia/delete_core_instance.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/calyptia/api/types"
+	"github.com/calyptia/cli/k8s"
+	"github.com/spf13/cobra"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func newCmdDeleteCoreInstance(config *config, testClientSet kubernetes.Interface) *cobra.Command {
+	var aggregatorName string
+	var skipError bool
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	configOverrides := &clientcmd.ConfigOverrides{}
+	cmd := &cobra.Command{
+		Use:   "instance",
+		Short: "Delete a core instance",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			if aggregatorName == "" {
+				return fmt.Errorf("the core instance name must be provided")
+			}
+			agg, err := config.cloud.Aggregators(ctx, config.projectID, types.AggregatorsParams{
+				Name: &aggregatorName,
+			})
+			if err != nil {
+				return err
+			}
+			aggregator := agg.Items[0]
+
+			if configOverrides.Context.Namespace == "" {
+				configOverrides.Context.Namespace = apiv1.NamespaceDefault
+			}
+
+			var clientset kubernetes.Interface
+			if testClientSet != nil {
+				clientset = testClientSet
+			} else {
+				kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+				kubeClientConfig, err := kubeConfig.ClientConfig()
+				if err != nil {
+					return err
+				}
+
+				clientset, err = kubernetes.NewForConfig(kubeClientConfig)
+				if err != nil {
+					return err
+				}
+
+			}
+
+			k8sClient := &k8s.Client{
+				Interface:    clientset,
+				Namespace:    configOverrides.Context.Namespace,
+				ProjectToken: config.projectToken,
+				CloudBaseURL: config.baseURL,
+			}
+
+			if err := k8sClient.EnsureOwnNamespace(ctx); err != nil {
+				return fmt.Errorf("could not ensure kubernetes namespace exists: %w", err)
+			}
+
+			label := "aggregatorID=" + aggregator.ID
+
+			deployments, err := k8sClient.FindDeploymentsByLabel(ctx, label)
+			if err != nil {
+				return err
+			}
+
+			for _, item := range deployments.Items {
```

Comment:
Isn't it possible to directly call delete without listing them first 🤔

Reply:
There is a `DeleteCollection` method that should do it.

Reply:
i couldnt find a way to delete by "tag" so im listing to get the health-check name, so bcz i have too do it for the health-check, im getting the deployment also